### ISSUE TITLE
How to find old development documentation

### ISSUE THREAD

**Issue by SilurianYang** [puppeteer/puppeteer#8936]
## Question

> Current documentation:
1. Unable to intuitively query the classification, it is too confusing
2. The quick classification is not clear and cannot be located immediately

**Comment by OrKoN** [comment]
Hey, sorry about the bad experience with the docs. Just to clarify do you refer to the https://pptr.dev (there is a search for the API documentation) or do you refer to the markdown version of the docs? The old doc website is preserved here https://puppeteer.github.io/pptr.dev/#?product=Puppeteer&version=v15.2.0 and also you can find the markdown documentation for older version by browsing tags (we have tagged every past release).

**Comment by SilurianYang** [comment]
hi, thanks for your reply! A very well-received feeling comes back when I open old documents, and the large shortcut menu on the left is really nice. Navigate to the description and parameter construction of its API through the first-level menu on the left, so that people can get what they want in a very short time.

**Comment by SilurianYang** [comment]
@OrKoN hi, thanks for your reply! A very well-received feeling comes back when I open old documents, and the large shortcut menu on the left is really nice. Navigate to the description and parameter construction of its API through the first-level menu on the left, so that people can get what they want in a very short time.

**Comment by OrKoN** [comment]
There is also the menu on the right in the new doc. Would the following feature https://github.com/puppeteer/puppeteer/issues/8897 simplify working with the API docs for you?

**Comment by SilurianYang** [comment]
Yes, I think the left menu needs to be streamlined and effectively categorized.

Find `puppeteer.launch()` vs:


####  1. The old documentation cost is almost zero and has detailed parameter descriptions

![1](https://user-images.githubusercontent.com/31280665/189600462-b4f4f0b6-45f3-4804-8fc1-f02d92864b19.gif)

####  2.  It's time consuming for me to find stuff in new documentation unless I know the documentation and API well

![2](https://user-images.githubusercontent.com/31280665/189604294-373d7c5b-3269-4d41-b398-4d0756887c3f.gif)

**Comment by markcellus** [comment]
I agree that the new documentation site is much less intuitive than the older one. 

I can never find the default values for things. For instance, the [WaitForOptions page](https://pptr.dev/next/api/puppeteer.waitforoptions) shows the options and talks about how to change the defaults but never tells you what the defaults are. Or the [`page.waitForNetworkIdle()`](https://pptr.dev/api/puppeteer.page.waitfornetworkidle) page, which doesn't tell you what the default values for `idleTime` and `timeout` are.

There are also pages that have very good descriptions of methods ([`page.waitForSelector()`](https://pptr.dev/api/puppeteer.page.waitforselector)), but then there are pages with no description of what it does ([`page.waitForFrame()`](https://pptr.dev/api/puppeteer.page.waitforframe)).

**Comment by OrKoN** [comment]
We now have Algolia search and default values are generated whenever possible. The sidebar is now also restructured. I am closing this one. Some of the points by @markcellus have to be fixed on the case by case basis (happy to review PRs) and some have been already fixed.

**Comment by markcellus** [comment]
@OrKoN 

Yes--some of the things I've pointed out have been addressed. Thanks for that!

However, I'd urge the primary maintainers to go through each documentation page and create issues related to the ones I've pointed out. Asking developers to contribute to documentation to a codebase they dont know much about probably isnt as productive. After all, we're asking for more documentation because we dont know what the code is supposed to do. 😅
